### PR TITLE
Ins 85 proxy generator

### PR DESCRIPTION
**Pull request by Vany**
codegenerator now supports generating wrappers for call contracts without reflection.

Makefile and tests in goplugin works.

### CODE REVIEW

File: logicrunner/goplugin/preprocessor/main.go [hunk 1]
```diff
@@ -71,39 +111,84 @@ func getMethods(F *ast.File) {
...
 			typename := r.(*ast.Ident).Name
-			methods[typename] = append(methods[typename], td)
+			ci.Methods[typename] = append(ci.Methods[typename], td)
 		}
 	}
+	return &ci
 }
 
-func generateWrappers() string {
+func generateTypes(ci *ContractInterface) string {
```
Comment (golangci) [inline logicrunner/goplugin/preprocessor/main.go]: `generateTypes` is unused

File: logicrunner/goplugin/preprocessor/main.go [hunk 2]
```diff
@@ -1,67 +1,107 @@
...
+	}
+
 	for _, fn := range flag.Args() {
-		w := generateForFile(fn)
-		_, _ = io.Copy(os.Stdout, w)
+		w, err := generateForFile(fn)
+		if err != nil {
+			panic(err)
+		}
+		io.Copy(output, w)
```
Comment (golangci) [inline logicrunner/goplugin/preprocessor/main.go]: Error return value of `io.Copy` is not checked

File: logicrunner/goplugin/preprocessor/main.go [hunk 3]
```diff
@@ -1,67 +1,109 @@
...
+
 	for _, fn := range flag.Args() {
-		w := generateForFile(fn)
-		_, _ = io.Copy(os.Stdout, w)
+		w, err := generateForFile(fn)
+		if err != nil {
+			panic(err)
+		}
+		if _, err = io.Copy(output, w); err != nil {
+			panic("Something weird with output: ", err)
```
Comment (golangci) [inline logicrunner/goplugin/preprocessor/main.go]: invalid operation: too many arguments for panic("Something weird with output: ", err) (expected 1, found 2)

File: logicrunner/goplugin/testplugins/primary/primary.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+// @inscontract
+// nolint
+type HwRunner struct {
+	Runned int
+}
+
+func (h *HwRunner) Run() string {
+	hw := Hw.GetNewInstance(logicrunner.Reference("#1.#2"))
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/primary/primary.go]: cannot use logicrunner.Reference("#1.#2") (constant "#1.#2" of type github.com/insolar/insolar/logicrunner.Reference) as Hw value in argument to Hw.GetNewInstance

File: logicrunner/goplugin/testplugins/primary/primary.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+type Hw struct {
+	Reference logicrunner.Reference
+}
+
+func (Hw) GetNewInstance(r logicrunner.Reference) Hw {
+	return Hw{Reference: r}
+}
+
+func (_self *Hw) Echo(s string) string {
+	foundation.APICall(_self.Reference, "Echo", s)
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/primary/primary.go]: too many arguments

File: logicrunner/goplugin/testplugins/primary/primary.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+}
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/primary/primary.go]: missing return

File: logicrunner/goplugin/preprocessor/main_test.go [hunk 7]
```diff
@@ -13,7 +11,7 @@ func Test_generateForFile(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	io.Copy(os.Stdout, w)
+	// io.Copy(os.Stdout, w)
```
Comment (ruz) [inline logicrunner/goplugin/preprocessor/main_test.go]: У тебя version control system - комментировать код не нужно :))))

File: logicrunner/goplugin/testplugins/foundation/foundation.go [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+	Unmarshal(interface{}, []byte)
+}
+
+// Call other contract via network dispatcher
+func Call(Reference logicrunner.Reference, MethodName string, Arguments []interface{}) ([]interface{}, error) {
+	return nil, nil
+}
+
+func APICall() { // GetPulsar / GetNodeList / GetValidatorCandidates
+	return
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/foundation/foundation.go]: redundant return statement

File: logicrunner/goplugin/testplugins/secondary/main.go [hunk 9]
```diff
@@ -1,12 +1,26 @@
...
+	"fmt"
+)
 
+// @inscontract
 // nolint
 type HelloWorlder struct {
 	Greeted int
 }
 
+type FullName struct {
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/secondary/main.go]: type FullName is unused

File: logicrunner/goplugin/testplugins/secondary/main.go [hunk 10]
```diff
@@ -1,12 +1,26 @@
...
+	First string
+	Last  string
+}
+
+type PersonalGreeting struct {
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/secondary/main.go]: type PersonalGreeting is unused

File: logicrunner/goplugin/testplugins/primary/primary.go [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+// @inscontract
+// nolint
+type HwRunner struct {
+	Runned int
+}
+
+//func (h *HwRunner) Run() string {
+//	hw := Hw.GetNewInstance(logicrunner.Reference("#1.#2"))
+//	return hw.Echo("Ooops")
+//}
```
Comment (ruz) [inline logicrunner/goplugin/testplugins/primary/primary.go]: Закоментированый код

File: logicrunner/goplugin/testplugins/secondary/main.go [hunk 12]
```diff
@@ -1,12 +1,28 @@
...
 }
 
+// nolint
+type FullName struct {
+	First string
+	Last  string
+}
+
+// molint
+type PersonalGreeting struct {
```
Comment (golangci) [inline logicrunner/goplugin/testplugins/secondary/main.go]: type PersonalGreeting is unused

File: logicrunner/goplugin/testplugins/primary/primary.go [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+func (Hw) GetNewInstance(r logicrunner.Reference) Hw {
+	return Hw{Reference: r}
+}
+
+//func (_self *Hw) Echo(s string) string {
+//	foundation.APICall(_self.Reference, "Echo", s)
+//	return s
+//}
```
Comment (ruz) [inline logicrunner/goplugin/testplugins/primary/primary.go]: И опять мусорный код

File: logicrunner/goplugin/preprocessor/main_test.go (hunk 7 above)
Comment (Vany) [reply logicrunner/goplugin/preprocessor/main_test.go]: Этот код хорошо иметь там для отладки.

**Pull request merged**